insights/core/dr.py

```diff
@@ -428,18 +428,29 @@ def get_registry_points(component):
         (set): A list of the registry points found.
     """
     reg_points = set()
-
+    # Process RegistryPoint spec of Specs
     if is_registry_point(component):
         reg_points.add(component)
-    else:
-        for dep in get_dependents(component):
```

Comment:
@xiangce yes, looks like you are right, `get_dependents` is moving in the wrong direction in the graph.